apps/nowcasting-app/components/helpers/utils.test.ts

```diff
@@ -181,3 +182,74 @@ describe("prettyPrintDayLabelWithDate", () => {
     expect(result).toBe("Today");
   });
 });
+
+describe("check y-axis max value calculation", () => {
+  test("should return 0 when chart data is null", () => {
+    const result = utils.calculateChartYMax(null);
+    expect(result).toBe(0);
+  });
+
+  test("should return 0 when chart data is undefined", () => {
+    const result = utils.calculateChartYMax(undefined);
+    expect(result).toBe(0);
+  });
+
+  test("should return 0 when chart data is empty array", () => {
+    const result = utils.calculateChartYMax([]);
+    expect(result).toBe(0);
+  });
+
+  test("should round up to nearest 2000 with buffer of 1000", () => {
+    const chartData = [{ value: 12500 }];
+    const result = utils.calculateChartYMax(chartData);
+    expect(result).toBe(14000); // 12500 + 1000 = 13500 rounded up to nearest 2000, ie 14000
+  });
+
+  test("should handle multiple numeric values", () => {
+    const chartData = [
+      { value1: 5000, value2: 8000 },
+      { value1: 3000, value2: 7000 }
+    ];
+    const result = utils.calculateChartYMax(chartData);
+    expect(result).toBe(10000); // (8000 + 1000) rounded up to nearest 2000
+  });
+
+  test("should ignore formattedDate fields", () => {
+    const chartData = [{ value: 6000, formattedDate: 15000 }];
```

Comment:
Nice, good to check this – this should be caught if you implement the concrete Type, but formattedDate I think is a `string`